src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -77,6 +101,28 @@ module _ {a b} {A : Set a} {B : Set b} (f : A → B) where
   map⁺ (swap x y p)  = swap _ _ (map⁺ p)
   map⁺ (trans p₁ p₂) = trans (map⁺ p₁) (map⁺ p₂)
 
+  -- permutations preserve 'being a mapped list'
+  map⁻ : ∀ {xs ys} → map f xs ↭ ys → ∃ λ ys′ → ys ≡ map f ys′
+  map⁻ {[]}     ρ rewrite ↭-[] (↭-sym ρ)  = -, refl
+  map⁻ {x ∷ []} ρ rewrite ↭-one (↭-sym ρ) = -, refl
+  map⁻ {_ ∷ _ ∷ _} refl           = -, refl
```

Comment:
Do we really need these rewrites or could we literally just paste the
proofs instead of `refl`.